### ISSUE TITLE
Sorting posts by current query then limiting no. of posts shown

### ISSUE THREAD

**Issue by gdoddsldn** [dudaster/ele-custom-skin#197]
Hi,

I'm trying to use the Pro custom skins on a category listing, and am using Elementor's theme builder in order to manage this. The plugin works nicely until I filter the results by 'Current Query', then it seems to ignore the 'Number of posts to show' option.

Example: https://s32182.p980.sites.pressdns.com/category/finance/ - here the top section should only have 1 visible post.

**Comment by dudaster** [comment]
Does the classic skin behaves the same?

**Comment by gdoddsldn** [comment]
No, all works as normal there. Only happens when using the plugin loops.
________________________________

**Comment by dudaster** [comment]
What about using a simple loop template?

**Comment by gdoddsldn** [comment]
I’ve just updated the page to a loop that just includes the title, the problem still persists.

**Comment by dudaster** [comment]
I've just pushed a new version out. Please update and let me know if the issue is fixed.

**Comment by gdoddsldn** [comment]
Unfortunately it’s still happening.

Cleared the cache etc. after installing the update.

**Comment by dudaster** [comment]
Can you provide admin access to a staging website? If yes please use the website chat to set it up.

**Comment by dudaster** [comment]
It seems when you use Current Query the number of posts cannot be set inside the Post Widgets. You can only set it in Wordpress Settings > Reading

But fortunately you can use Elementor custom query to make a hook and filter the current query: https://developers.elementor.com/custom-query-filter/

In your case you case you can use this code:
```
// change the number of posts per page in elementor qustom query
add_action( 'elementor/query/my_pagination', function( $query ) {
	$query->set( 'posts_per_page', 1 );
} );
```
in your child theme's function.php .

And add it in your query tab:
![Screen Shot 2020-05-04 at 5 55 08 PM](https://user-images.githubusercontent.com/8319178/80981287-71faeb00-8e32-11ea-8f27-df50b3b4d9f8.png)

**Comment by turbit-smart** [comment]
wow, thanks your small tip helped me big time,
I just changed the number in here and all fixed Wordpress Settings > Reading
before couldn't' control the amount throw the widget
